core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -477,12 +605,16 @@ private[spark] class MapOutputTrackerWorker(conf: SparkConf) extends MapOutputTr
 private[spark] object MapOutputTracker extends Logging {
 
   val ENDPOINT_NAME = "MapOutputTracker"
+  private val DIRECT = 0
+  private val BROADCAST = 1
 
   // Serialize an array of map output locations into an efficient byte format so that we can send
   // it to reduce tasks. We do this by compressing the serialized bytes using GZIP. They will
   // generally be pretty compressible because many map outputs will be on the same hostname.
-  def serializeMapStatuses(statuses: Array[MapStatus]): Array[Byte] = {
+  def serializeMapStatuses(statuses: Array[MapStatus], broadcastManager: BroadcastManager,
+      isLocal: Boolean, minBroadcastSize: Int): (Array[Byte], Broadcast[Array[Byte]]) = {
```

Comment:
Good point, I missed that :-)

On Friday, April 1, 2016, Tom Graves <notifications@github.com> wrote:

> In core/src/main/scala/org/apache/spark/MapOutputTracker.scala
> <https://github.com/apache/spark/pull/12113#discussion_r58253255>:
>
> >
> >    // Serialize an array of map output locations into an efficient byte format so that we can send
> >    // it to reduce tasks. We do this by compressing the serialized bytes using GZIP. They will
> >    // generally be pretty compressible because many map outputs will be on the same hostname.
> > -  def serializeMapStatuses(statuses: Array[MapStatus]): Array[Byte] = {
> > +  def serializeMapStatuses(statuses: Array[MapStatus], broadcastManager: BroadcastManager,
> > +      isLocal: Boolean, minBroadcastSize: Int): (Array[Byte], Broadcast[Array[Byte]]) = {
>
> Both values are used in getSerializedMapOutputStatuses
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/12113/files/3c1def02b80ebfed55904e504609aa02de6559ce#r58253255>
>